mujoco_warp/_src/solver.py

```diff
@@ -101,18 +103,39 @@ def _efc_kernel(d: types.Data):
     Jaref = d.efc.Jaref[efcid]
     efc_D = d.efc.D[efcid]
 
-    # TODO(team): active and conditionally active constraints
-    active = int(Jaref < 0.0)
-    d.efc.active[efcid] = active
+    active = (Jaref < 0.0) or (efcid < d.ne[0] + d.nf[0])
+
+    if wp.static(m.dof_frictionloss_adr.size > 0 and (not DSBL_FLOSS)):
+      f = d.efc.frictionloss[efcid]
+      r = _safe_div(1.0, efc_D)
+      rf = r * f
+      f_pos = f > 0.0
+      linear_neg = (Jaref <= -rf) and f_pos
```

Comment:
This looks like a JAX-y no-if version of the code.  I'm thinking Warp can let us do something a bit more readable:

```py
if Jaref <= -rf:
  cost = -0.5 * rf * f - f * Jaref
  efc_force = f
elif Jaref >= rf:
  ...
```

Benchmark it though to make sure it doesn't slow down the step!

Reply:
updated the code to use if statements